src/vsr/journal.zig

```diff
@@ -706,6 +717,9 @@ pub fn JournalType(comptime Replica: type, comptime Storage: type) type {
         ) void {
             assert(journal.status == .recovered);
             assert(checksum != 0);
+            if (destination_replica == null) {
+                assert(journal.reads.available() > 0);
```

Comment:
Or maybe `journal.reads.available() >= reads_commit_count_max`? Effectively the same thing, but I think it a little more self-explanatory.

Reply:
should be

```
assert(journal.reads.available() >= reads_commit_count_max - reads_commit_count);
```

right?

Reply:
I think it is the latter outside of `destination_replica == null`, or it is the former inside. I think I prefer your version, to avoid the if block.

Reply:
Hm, I think we are on different pages so far. I think this is what is fully explicit here: 

```
            if (destination_replica == null) {
                assert(journal.reads.available() >=
                    reads_commit_count_max - journal.reads_commit_count);
            }
```

"If we are on the commit path, we have enough commit reads available sans already in-progress commit"

The `journal.reads.available() >= reads_commit_count_max` version would be wrong I think? You might spend `< reads_commit_count_max` reads.

Now that I look at it, I think I like 

```
            if (destination_replica == null) {
                assert(journal.reads.available() > 0);
```

most actually


We are not asserting that the count is precise (there are other assertions for that). Rather we assert that a commit path cant' be dropped due to lack of iops. Will switch back to that

Reply:
> Which is not what we want?

Yes. There are two things we might want to assert: 

* that the counts are right -- we have at least max - count reads available (this is true 
 for both the commit and the repair path)
* that we have a spare read (this is true for commit path, but not for repair path)


```
                assert(journal.reads.available() >=
                    reads_commit_count_max - journal.reads_commit_count);
```

is the assert for the frist point, the `> 0` is for the second point. 

Now that I think about it, the appropriate generalization would actually be

```
assert(journal.reads_commit_count < reads_commit_count_max)
```

but `available > 0` still feels like a harder to confuse thing.

Reply:
Makes sense, thanks!